code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -350,15 +325,21 @@
 	S.key = T.key
 	if(U)
 		S.faction |= "\ref[U]" //Add the master as a faction, allowing inter-mob cooperation
-	if(U && iscultist(U))
+		if(!iscultist(U))
+			S.mind.store_memory("<b>Serve [U.real_name], your creator.</b>")
+	if(iscultist(U))
 		SSticker.mode.add_cultist(S.mind, 0)
+		S.mind.special_role = SPECIAL_ROLE_CULTIST
+	if(iswizard(U))
+		SSticker.mode.update_wiz_icons_added(S.mind)
+		S.mind.special_role = SPECIAL_ROLE_WIZARD_APPRENTICE
 	S.cancel_camera()
 	name = "soulstone: Shade of [T.real_name]"
 	icon_state = "soulstone2"
 	if(U && iswizard(U))

```

Comment:
These two `if`s can also be folded to be under the overall `if(U)` check. The three unconditional lines above can go either above or below the entire `if(U)` block - there's no particular need for them to be here in the middle.

Reply:
I finally see what you're saying, I'll adjust.